doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+>
+> _other-new-line_ ::= U+0085 NEXT LINE | U+2028 LINE SEPARATOR | U+2029 PARAGRAPH SEPARATOR
+
+#### Token value
+
+A ***comment*** token has no value.
+
+#### Notes
+
+A _comment-character_ is any character that is not the first character of _new-line_. New line characters and sequences correspond to the [line boundaries requirement RL1.6 of Unicode Regular Expressions](https://www.unicode.org/reports/tr18/#RL1.6).
+
+
+### Whitespace
+<a id="whitespace"/>
+
+In some expressions, where it denotes multiplication, whitespace is mandatory, but otherwise whitespace is not significant. It comprises any non-empty sequence of characters with the White_Space property.
+
+> ***whitespace*** ::= {White_Space}\+
+
+#### Token value
+
+A ***whitespace*** token has no value.
+
+
+### Symbols
+<a id="symbols"/>
+
+Symbols correspond to Arblang identifiers and keywords. Arblang keywords are generally contextual: the interpretation of a symbol as an identifier or a keyword is determined in parsing.
+
+> ***symbol*** ::= _symbol-start_ _symbol-continue_\* | _symbol-degree-prefixed_
+>
+> _symbol-start_ ::= {L} | {Nl} | {Other_ID_Start}
+>
+> _symbol-continue_ ::= _symbol-start_ | {Nd} | {Mn} | {Mc} | {Pc} | _prime-mark_
+>
+> _prime-mark_ ::= U+0027 `'` APOSTROPHE | U+02B9 `ʹ` MODIFIER LETTER PRIME | U+2032 `′` PRIME
+>
+> _symbol-degree-prefixed_ ::= U+00B0 `°` DEGREE SIGN {L}+ | U+2109 `℉` DEGREE FAHRENHEIT | U+2109 `℃` DEGREE CENTIGRADE
+
+#### Token value
+
+Canonicalization substitutes any character in _prime-mark_ with U+0027 APOSTROPHE after performing NFKC normalization.
+
+#### Notes
+
+Identifiers broadly follow the conventions of Python, except:
+* `_` is not a valid _symbol-start_.
+* A ***symbol*** may contain _prime-mark_ characters.
+* Special degree-sign prefixed symbols are permitted, in order to represent temperature scales. Compatibility characters U+2109 `℉` and U+2109 `℃` will be converted to `°F` and `°C` respectively in NFKC normalization.
+* Characters that are only _compatibility_ equivalent to an acceptable _symbol_ character are not permitted. In particular, superscript numerals are not permitted within a _symbol_.
+
+
+### Superscript literals
+<a id="superscript-literals"/>
+
+A superscript literal is used to denote exponents in the absence of the exponentiation operator.
+
+> ***superscript-literal*** ::= _superscript-minus_? _superscript-digit_+
+>
+> _superscript-minus_ ::= U+207B `⁻` SUPERSCRIPT MINUS
+>
+> _superscript-digit_ ::= `⁰` | `¹` | `²` | `³` | `⁴` | `⁵` | `⁶` | `⁷` | `⁸` | `⁹`
+
+#### Token value
+
+After NFKC normalization, which takes superscript digits and minus to regular digits and U+2212 MINUS SIGN, the token value is canonicalized by mapping U+2212 to U+002d `-` HYPHEN-MINUS.
+
+
+### Numeric literals
+<a id="numeric-literals"/>
+
+Numeric literals represent both fractional and integer literal values.
+
+> ***numeric-literal*** ::= _digit-sequence_ (_decimal-separator_ _digit-sequence_)? _exponent_?
```

Comment:
No sign here?

Reply:
No, but only so as to simplify the scanning:
* `a-3` currently scans as _**symbol**_ _**minus-sign**_ _**numeric-literal**_, which will be parsed as an arithmetic expression.
* If we include an optional sign prefix to a numeric literal, there is a lexical ambiguity that could not be resolved without more context or some sort of rule precedence, the alternate tokenization being _**symbol**_ _**numeric-literal**_, which would not parse.

Reply:
So, no `- -5`?

Reply:
`- -5` gets tokenized as ***minus-sign*** ***minus-sign*** ***numeric-literal***, and by one of the expression syntax rules,

>_algebraic-factor_ ::= _quantity-literal_ | ( ***minus-sign*** | ***square-root*** )* _algebraic-base_ _algebraic-exponent_? )

This will get parsed as two minuses, with the precedence rules and semantics saying that this means two unary minuses applied to the number 5.